### PR TITLE
feat: Fetch adapter support for `context` provided via adapterOptions

### LINKED ISSUE #56: Add ability to configure window in fetch adapter.
In E2E frameworks like [cypress](https://www.cypress.io/) `fetch` is attached to an injected window object like this:

```js
cy.visit('my/pathname/', {
  onBeforeLoad: win => { ... }
})
```

Currently the fetch adapter expects `fetch` to be attached to `self` but in this case it's not so it would be nice if I could supply the window object to the fetch adapter or Polly constructor. E.g.
```js
cy.visit('my/pathname/', {
  onBeforeLoad: win => {
    const polly = new Polly('my-polly-recording', {
      adapters: [['fetch', FetchAdapter.create({ window: win })]]
    });
  }
})
```

### PR DESCRIPTION
**Pull request by jasonmit**
I'll update the docs also.

I can't  apply the same treatment to the XHR adapter since `nise` doesn't enable providing a context object.  I think we spoke about this already, but mentioning again.

Resolves #56

### CODE REVIEW

File: packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js [hunk 1]
```diff
@@ -21,3 +21,36 @@ describe('Integration | Fetch Adapter', function() {
...
+
+describe('Integration | Fetch Adapter | Concurrency', function() {
+  it('should prevent concurrent fetch adapter instances', function() {
+    const one = new Polly('one');
+    const two = new Polly('two');
+
+    one.connectTo(FetchAdapter);
+
+    expect(function() {
+      two.connectTo(FetchAdapter);
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js]: `connectTo` vs providing the adapter via `configure` so that the constructor assignment happens and we can do cleanup/`stop()`.

File: packages/@pollyjs/adapter-fetch/src/index.js [hunk 2]
```diff
@@ -1,34 +1,46 @@
...
+    return {
+      context: global
+    };
+  }
+
   onConnect() {
-    this.assert('Fetch global not found.', nativeFetch);
+    const { context } = this.options;
+
+    this.assert('Fetch global not found.', context.fetch);
```
Comment (offirgolan) [inline packages/@pollyjs/adapter-fetch/src/index.js]: `!!(context && context.fetch)` 

Assert only checks for false or undefined. Might wanna open that up to any falsey value.

File: packages/@pollyjs/adapter-fetch/src/index.js [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
+    context.fetch = (url, options = {}) =>
       this.handleRequest({
         url,
         method: options.method || 'GET',
         headers: FetchUtils.serializeHeaders(options.headers),
         body: options.body,
         requestArguments: [url, options]
       });
+
+    defineProperty(context.fetch, STUB_META, { value: {} });
```
Comment (offirgolan) [inline packages/@pollyjs/adapter-fetch/src/index.js]: Why a meta object vs a Boolean?

File: packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js [hunk 4]
```diff
@@ -21,3 +21,36 @@ describe('Integration | Fetch Adapter', function() {
   adapterTests();
   adapterBrowserTests();
 });
+
+describe('Integration | Fetch Adapter | Concurrency', function() {
```
Comment (offirgolan) [inline packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js]: Can you also add a test against a null context and one without fetch?

**Comment by offirgolan** [comment]
> I can't apply the same treatment to the XHR adapter 

I just meant the logic around checking the STUB_META instead of checking against the captured native class in the module. That way, we can still wrap a pre modified XHR.

File: packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js (hunk 4 above)
Comment (jasonmit) [reply packages/@pollyjs/adapter-fetch/tests/integration/adapter-test.js]: Yup need to add tests in general around context.  Will finish up tonight + docs

**Comment by jasonmit** [comment]
> I just meant the logic around checking the STUB_META instead of checking against the captured native class in the module. That way, we can still wrap a pre modified XHR.

I'll tackle in a separate PR tomorrow.
